classes/file/FileManager.inc.php

```diff
@@ -540,6 +540,11 @@ function parseFileExtension($fileName) {
 			$fileExtension = 'txt';
 		}
 
+		// consider .tar.gz extension
+		if (strcasecmp($fileExtension, 'gz') == 0 && strcasecmp($fileParts[count($fileParts) - 2],  'tar') == 0) {
```

Comment:
I think

    if (strtolower(substr($filename, -7)) == '.tar.gz') { ...

...is more readable, and won't cause warnings -- OTOH if you upload a `.gz` file that's not a `.tar.gz` you'll get negative indexes.